Convert `build_bundle.js` to use CommonJS

eddyashton opened the pull request:
The `import` syntax was only recently supported in `node` (v12), so does not work in the node that is available in our Docker containers (v10). To avoid compile errors, I've rewritten it to use the older `require` syntax, so it should work across more `node` versions.

jumaffre commented:
Would it be better to add a later version of node to our `ccf-app-ci` container image instead? Just like we do for `ccf-ci`. If we're expecting developers to build JS apps using the `ccf-app-ci` image, I guess we should use the latest type of includes there too?

eddyashton commented:
> Would it be better to add a later version of node to our `ccf-app-ci` container image instead? Just like we do for `ccf-ci`. If we're expecting developers to build JS apps using the `ccf-app-ci` image, I guess we should use the latest type of includes there too?

I think we should do that as well/separately, but I think this fix is independently useful - even if people aren't using our Docker image, this tiny util script should be useable in most npm environments.

jumaffre commented:
> > Would it be better to add a later version of node to our `ccf-app-ci` container image instead? Just like we do for `ccf-ci`. If we're expecting developers to build JS apps using the `ccf-app-ci` image, I guess we should use the latest type of includes there too?
> 
> I think we should do that as well/separately, but I think this fix is independently useful - even if people aren't using our Docker image, this tiny util script should be useable in most npm environments.

I agree for the util script. I'm also thinking of the consequences of setting `"type": "commonjs"` in `package.json`. The sample there is TypeScript anyway, so maybe that's fine?

eddyashton commented:
> > > Would it be better to add a later version of node to our `ccf-app-ci` container image instead? Just like we do for `ccf-ci`. If we're expecting developers to build JS apps using the `ccf-app-ci` image, I guess we should use the latest type of includes there too?
> > 
> > 
> > I think we should do that as well/separately, but I think this fix is independently useful - even if people aren't using our Docker image, this tiny util script should be useable in most npm environments.
> 
> I agree for the util script. I'm also thinking of the consequences of setting `"type": "commonjs"` in `package.json`. The sample there is TypeScript anyway, so maybe that's fine?

Yeah I've reverted this change and replaced it with a version dependency on node >=14. I think trying to support v10 is a waste of time (this was only adding support for a single util script, the remainder of our samples still had cascading errors). There's still the question of whether we should be installing 14 in our `app-ci` images, but I think not - at the moment it's used for CI of C++ apps, which don't want an unnecessary node dependency. If we think `app-dev` includes JS `app-dev`, then we should install 'our' version of node there.
